stream.go

```diff
@@ -161,6 +173,16 @@ func (cs *clientStream) SendMsg(m interface{}) (err error) {
 
 func (cs *clientStream) RecvMsg(m interface{}) (err error) {
 	err = recv(cs.p, cs.codec, m)
+	defer func() {
+		// err != nil indicates the termination of the stream.
+		if EnableTracing && err != nil {
+			cs.traceInfo.tr.Finish()
```

Comment:
This should have been discovered by the manual testing. We were lazy and reused the benchmark binary to test this. But unfortunately the benchmark only performs streaming rpcs whose lifetime is as same as the binary itself so that this bug was hidden. Please revise it and make sure it works properly.